doubly_linked_list/doubly_linked_list.py

```diff
@@ -1,68 +1,170 @@
 """Each ListNode holds a reference to its previous node
 as well as its next node in the List."""
+
+
 class ListNode:
-  def __init__(self, value, prev=None, next=None):
-    self.value = value
-    self.prev = prev
-    self.next = next
+    def __init__(self, value, prev=None, next=None):
+        self.value = value
+        self.prev = prev
+        self.next = next
 
-  """Wrap the given value in a ListNode and insert it
+    """Wrap the given value in a ListNode and insert it
   after this node. Note that this node could already
   have a next node it is point to."""
-  def insert_after(self, value):
-    current_next = self.next
-    self.next = ListNode(value, self, current_next)
-    if current_next:
-      current_next.prev = self.next
 
-  """Wrap the given value in a ListNode and insert it
+    def insert_after(self, value):
+        current_next = self.next
+        self.next = ListNode(value, self, current_next)
+        if current_next:
+            current_next.prev = self.next
+
+    """Wrap the given value in a ListNode and insert it
   before this node. Note that this node could already
   have a previous node it is point to."""
-  def insert_before(self, value):
-    current_prev = self.prev
-    self.prev = ListNode(value, current_prev, self)
-    if current_prev:
-      current_prev.next = self.prev
 
-  """Rearranges this ListNode's previous and next pointers
+    def insert_before(self, value):
+        current_prev = self.prev
+        self.prev = ListNode(value, current_prev, self)
+        if current_prev:
+            current_prev.next = self.prev
+
```

Comment:
Kind of cool that all you have to do is mess around with the points behind and in front of the node, and it effectively changes the whole structure of the linked list. It's an amazing conceit!